base/tests/freetrm/eventstore/DeDuplicatorTest.scala

```diff
@@ -0,0 +1,188 @@
+package freetrm.eventstore
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.stream.scaladsl.Source
+import akka.util.Timeout
+import freetrm.eventstore.DeDuplicator.{WriteResult, SingleTopicDeDuplicator}
+import freetrm.eventstore.http.ESTestUtils.TestEventSourceReader
+import freetrm.eventstore.http.FunSuiteDoc
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatest.{FunSuite, Matchers}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Random, Try}
+
+class DeDuplicatorTest extends FunSuite with FunSuiteDoc 
+  with ScalaFutures with ScalatestRouteTest with Matchers with MockFactory {
+  implicit val defaultPatience = PatienceConfig(timeout = Span(2, Seconds), interval = Span(5, Millis))
+
+  documentTests(
+    "de-duplicates" testsThat
+      """
+        When you write duplicate events to the de-duplicator they don't get written through to the underlying writer. 
+      """,
+    "de dup only loads once per topic" testsThat
+      """
+        The de-duplicator only loads the history of events the first time you write to it.
+      """,
+    """will reload the deduplicator if it fails""" testsThat
+      """
+        The loading code will recover from failing to load the history of events by trying again on the next
+        write. The Future for the first write will indicate the failure.
+      """,
+    "the SingleTopicDeDuplicator is created on demand and contains the right events" testsThat
+      """
+        When you write the first event for a topic the de-duplicator is loaded. Also tests that
+        the events loaded are correct.
+      """
+  )
+  
+  
+  test("de-duplicates") {
```

Comment:
This isn't testing that when we call with the same data as the latest surface, but _with a different version no_, it doesn't get written